### PR TITLE
feat(cli): onboard iaas volume backup api 

### PR DESCRIPTION
**Pull request by Benjosh95**
## Description

Onboarding the volume backup API https://docs.api.stackit.cloud/documentation/iaas/version/v1#tag/Backups to manage the backup resources via the cli.

## Checklist

- [x] Issue was linked above
- [x] Code format was applied: `make fmt`
- [x] Examples were added / adjusted (see e.g. [here](https://github.com/stackitcloud/stackit-cli/blob/ef291d1683ca5b0d719ec0a26ecb999a32685117/internal/cmd/ske/cluster/create/create.go#L49-L63))
- [x] Docs are up-to-date: `make generate-docs` (will be checked by CI)
- [x] Unit tests got implemented or updated
- [x] Unit tests are passing: `make test` (will be checked by CI)
- [x] No linter issues: `make lint` (will be checked by CI) 

## Testing-Instructions

1. Create a test volume to have something to backup
stackit volume create --name backup-test-vol --size 10 --performance-class storage_premium_perf0 --availability-zone eu01-1 --labels initial=true,purpose=backup-testing 

2. Create backup of the volume with name and labels
stackit volume backup create --source-id <VOLUME_ID> --source-type volume --name first-backup --labels state=original,test=true

3. List backups with limit and label selector
stackit volume backup list --limit 5 --label-selector state=original

4. Update backup with new name and labels
stackit volume backup update <BACKUP_ID> --name updated-backup --labels state=modified,test=true

5. Describe backup to verify changes
stackit volume backup describe <BACKUP_ID>

6. Restore backup
stackit volume backup restore <BACKUP_ID> 

7. Clean up by deleting backup
stackit volume backup delete <BACKUP_ID>

8. Clean up by deleting volume
stackit volume delete <VOLUME_ID>

### CODE REVIEW

File: internal/cmd/volume/backup/create/create.go [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+		} else {
+			p.Debug(print.DebugLevel, "parsed input values: %s", modelStr)
+		}
+	}
+
+	return &model, nil
+}
+
+// TODO: Enough?
+func buildRequest(model *inputModel, apiClient *iaas.APIClient, ctx context.Context) iaas.ApiCreateBackupRequest {
```
Comment (rubenhoenle) [inline internal/cmd/volume/backup/create/create.go]: resolve todo

File: internal/cmd/volume/backup/create/create.go [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+	// TODO: doc says if createeBackup func provides snapshot-id but isnt in the func-signature?
+	req := apiClient.CreateBackup(ctx, model.ProjectId)
+	return req
+}
+
+// TODO: create(volume)BackupResponse or createBackupResponse needs to be created
```
Comment (rubenhoenle) [inline internal/cmd/volume/backup/create/create.go]: resolve todo

File: internal/cmd/volume/backup/create/create_test.go [hunk 3]
```diff
@@ -0,0 +1,287 @@
...
+	"github.com/stackitcloud/stackit-cli/internal/pkg/globalflags"
+	"github.com/stackitcloud/stackit-cli/internal/pkg/print"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"github.com/google/uuid"
+	"github.com/stackitcloud/stackit-sdk-go/services/iaas"
+)
+
+var projectIdFlag = globalflags.ProjectIdFlag
```
Comment (rubenhoenle) [inline internal/cmd/volume/backup/create/create_test.go]: I know this is done in a lot of places across the codebase, but please don't wrap the `globalflags.ProjectIdFlag` in another variable, instead just use `globalflags.ProjectIdFlag` directly please.

File: internal/cmd/volume/backup/list/list_test.go [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{
+			name:    "empty",
+			args:    args{},
+			wantErr: true,
+		},
+		{
```
Comment (rubenhoenle) [inline internal/cmd/volume/backup/list/list_test.go:209]: consider adding a test case which tests an empty iaas backup in the slice

```golang
args: args{
	backups: []iaas.Backup{{}},
},
```

File: internal/cmd/volume/backup/restore/restore_test.go [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+	"github.com/stackitcloud/stackit-cli/internal/pkg/globalflags"
+	"github.com/stackitcloud/stackit-cli/internal/pkg/print"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"github.com/google/uuid"
+	"github.com/stackitcloud/stackit-sdk-go/services/iaas"
+)
+
+var projectIdFlag = globalflags.ProjectIdFlag
```
Comment (rubenhoenle) [inline internal/cmd/volume/backup/restore/restore_test.go]: same here, don't wrap it with another variable pls

File: internal/cmd/volume/backup/update/update_test.go [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+	"github.com/stackitcloud/stackit-cli/internal/pkg/globalflags"
+	"github.com/stackitcloud/stackit-cli/internal/pkg/print"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"github.com/google/uuid"
+	"github.com/stackitcloud/stackit-sdk-go/services/iaas"
+)
+
+var projectIdFlag = globalflags.ProjectIdFlag
```
Comment (rubenhoenle) [inline internal/cmd/volume/backup/update/update_test.go]: same here

File: internal/cmd/volume/backup/create/create.go [hunk 7]
```diff
@@ -0,0 +1,234 @@
...
+		}
+	}
+
+	return &model, nil
+}
+
+func buildRequest(ctx context.Context, model *inputModel, apiClient *iaas.APIClient) iaas.ApiCreateBackupRequest {
+	req := apiClient.CreateBackup(ctx, model.ProjectId)
+
+	// Convert map[string]string to map[string]interface{}
```
Comment (rubenhoenle) [inline internal/cmd/volume/backup/create/create.go]: This code is duplicated already 20 times across the codebase: https://github.com/search?q=repo%3Astackitcloud%2Fstackit-cli%20%2F%2F%20Convert%20map%5Bstring%5Dstring%20to%20map%5Bstring%5Dinterface%7B%7D&type=code

Please create a util function for it. 

You can also create a Jira ticket in the backlog instead if you prefer. But we shouldn't continue to work like that.

File: internal/cmd/volume/backup/create/create_test.go [hunk 8]
```diff
@@ -0,0 +1,298 @@
...
+		backup       *iaas.Backup
+	}
+
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{
+			name:    "empty backup",
```
Comment (rubenhoenle) [inline internal/cmd/volume/backup/create/create_test.go:242]:
```suggestion
			name:    "backup is nil",
```

Comment (rubenhoenle) [reply internal/cmd/volume/backup/create/create_test.go:242]: Empty backup is another testcase which you should add IMO

File: internal/cmd/volume/backup/create/create.go [hunk 9]
```diff
@@ -0,0 +1,234 @@
...
+		details, err := yaml.MarshalWithOptions(resp, yaml.IndentSequence(true), yaml.UseJSONMarshaler())
+		if err != nil {
+			return fmt.Errorf("marshal backup: %w", err)
+		}
+		p.Outputln(string(details))
+		return nil
+
+	default:
+		if async {
+			p.Outputf("Triggered backup of %s in %s. Backup ID: %s\n", sourceLabel, projectLabel, *resp.Id)
```
Comment (rubenhoenle) [inline internal/cmd/volume/backup/create/create.go]: will panic if `resp.Id` is nil, see my comment below about the missing testcase

File: internal/cmd/volume/backup/list/list.go [hunk 10]
```diff
@@ -0,0 +1,194 @@
...
+			return fmt.Errorf("marshal backup list: %w", err)
+		}
+		p.Outputln(string(details))
+		return nil
+
+	default:
+		table := tables.NewTable()
+		table.SetHeader("ID", "NAME", "SIZE", "STATUS", "SNAPSHOT ID", "VOLUME ID", "AVAILABILITY ZONE", "LABELS", "CREATED AT", "UPDATED AT")
+
+		for i := range backups {
```
Comment (rubenhoenle) [inline internal/cmd/volume/backup/list/list.go]:
```suggestion
		for _, backup := range backups {
```

or did I miss something? :smile:

File: internal/cmd/volume/backup/list/list_test.go (hunk 4 above)
Comment (rubenhoenle) [reply internal/cmd/volume/backup/list/list_test.go:209]: Well, now you don't have a testcase for an empty slice anymore...

File: internal/cmd/volume/backup/restore/restore.go [hunk 11]
```diff
@@ -0,0 +1,149 @@
...
+			if backup != nil && backup.Name != nil {
+				backupLabel = *backup.Name
+			}
+
+			// Get source details for labels
+			var sourceLabel string
+			if backup != nil && backup.VolumeId != nil {
+				volume, err := apiClient.GetVolume(ctx, model.ProjectId, *backup.VolumeId).Execute()
+				if err != nil {
+					params.Printer.Debug(print.ErrorLevel, "get volume details: %v", err)
```
Comment (rubenhoenle) [inline internal/cmd/volume/backup/restore/restore.go:67]: the line `sourceLabel = *backup.VolumeId` is the same as in the else branch.
Just put this line before the whole if/else if/ else block. 

Then you can save yourself the duplication and the else block :)

File: internal/cmd/volume/backup/restore/restore.go [hunk 12]
```diff
@@ -0,0 +1,149 @@
...
+				s := spinner.New(params.Printer)
+				s.Start("Restoring backup")
+				_, err = wait.RestoreBackupWaitHandler(ctx, apiClient, model.ProjectId, model.BackupId).WaitWithContext(ctx)
+				if err != nil {
+					return fmt.Errorf("wait for backup restore: %w", err)
+				}
+				s.Stop()
+			}
+
+			projectLabel := model.ProjectId
```
Comment (rubenhoenle) [inline internal/cmd/volume/backup/restore/restore.go]: I don't see the point in creating a new variable here

File: internal/cmd/volume/backup/create/create.go [hunk 13]
```diff
@@ -0,0 +1,234 @@
...
+			examples.NewExample(
+				`Create a backup from a volume`,
+				"$ stackit volume backup create --source-id xxx --source-type volume --project-id xxx"),
+			examples.NewExample(
+				`Create a backup from a snapshot with a name`,
+				"$ stackit volume backup create --source-id xxx --source-type snapshot --name my-backup --project-id xxx"),
+			examples.NewExample(
+				`Create a backup with labels`,
+				"$ stackit volume backup create --source-id xxx --source-type volume --labels key1=value1,key2=value2 --project-id xxx"),
+		),
```
Comment (marceljk) [inline internal/cmd/volume/backup/create/create.go]: The CLI has also an option to configure the project-id global for all commands. The project-id can be overwritten in the commands with the flag, but for the examples I wouldn't add the flag to stay consistent with the other commands and their examples
```suggestion
			examples.NewExample(
				`Create a backup from a volume`,
				"$ stackit volume backup create --source-id xxx --source-type volume"),
			examples.NewExample(
				`Create a backup from a snapshot with a name`,
				"$ stackit volume backup create --source-id xxx --source-type snapshot --name my-backup"),
			examples.NewExample(
				`Create a backup with labels`,
				"$ stackit volume backup create --source-id xxx --source-type volume --labels key1=value1,key2=value2"),
		),
```

File: internal/cmd/volume/backup/create/create.go [hunk 14]
```diff
@@ -0,0 +1,234 @@
...
+		},
+	}
+
+	configureFlags(cmd)
+	return cmd
+}
+
+func configureFlags(cmd *cobra.Command) {
+	cmd.Flags().String(sourceIdFlag, "", "ID of the source from which a backup should be created")
+	cmd.Flags().String(sourceTypeFlag, "", "Source type of the backup (volume or snapshot)")
```
Comment (marceljk) [inline internal/cmd/volume/backup/create/create.go]: This can be configured with `flags.EnumFlag()`. See also: https://github.com/stackitcloud/stackit-cli/blob/4f7804790e9f3306c2d3ea770f674239bbaea3b9/internal/cmd/postgresflex/instance/update/update.go#L141

This as the advantage, that you can define the allowed values in a slice and reuse it also for the validation and description. And in the future it can easily be extended by just adding a value to the slice

File: internal/cmd/volume/backup/delete/delete.go [hunk 15]
```diff
@@ -0,0 +1,132 @@
...
+func NewCmd(params *params.CmdParams) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   fmt.Sprintf("delete %s", backupIdArg),
+		Short: "Deletes a backup",
+		Long:  "Deletes a backup by its ID.",
+		Args:  args.SingleArg(backupIdArg, utils.ValidateUUID),
+		Example: examples.Build(
+			examples.NewExample(
+				`Delete a backup`,
+				"$ stackit volume backup delete xxx-xxx-xxx"),
```
Comment (marceljk) [inline internal/cmd/volume/backup/delete/delete.go]: Just a nitpick to stay consistent with the existing commands
```suggestion
				`Delete a backup with ID "xxx"`,
				"$ stackit volume backup delete xxx"),
```

File: internal/cmd/volume/backup/delete/delete.go [hunk 16]
```diff
@@ -0,0 +1,132 @@
...
+			examples.NewExample(
+				`Delete a backup and wait for deletion to be completed`,
+				"$ stackit volume backup delete xxx-xxx-xxx --async=false"),
```
Comment (marceljk) [inline internal/cmd/volume/backup/delete/delete.go]: The async flag can be globally configured like the projectId and is on default already false. So compared to the previous example there is no difference.
I think it's fine to have just one example for the delete command. For other delete commands we also have only one example

File: internal/cmd/volume/backup/describe/describe.go [hunk 17]
```diff
@@ -0,0 +1,153 @@
...
+func NewCmd(params *params.CmdParams) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   fmt.Sprintf("describe %s", backupIdArg),
+		Short: "Describes a backup",
+		Long:  "Describes a backup by its ID.",
+		Args:  args.SingleArg(backupIdArg, utils.ValidateUUID),
+		Example: examples.Build(
+			examples.NewExample(
+				`Get details of a backup`,
+				"$ stackit volume backup describe xxx-xxx-xxx"),
```
Comment (marceljk) [inline internal/cmd/volume/backup/describe/describe.go]:
```suggestion
				`Get details of a backup with ID "xxx"`,
				"$ stackit volume backup describe xxx"),
```

File: internal/cmd/volume/backup/describe/describe.go [hunk 18]
```diff
@@ -0,0 +1,153 @@
...
+			examples.NewExample(
+				`Get details of a backup in JSON format`,
+				"$ stackit volume backup describe xxx-xxx-xxx --output-format json"),
```
Comment (marceljk) [inline internal/cmd/volume/backup/describe/describe.go]:
```suggestion
				`Get details of a backup with ID "xxx" in JSON format`,
				"$ stackit volume backup describe xxx --output-format json"),
```

File: internal/cmd/volume/backup/describe/describe.go [hunk 19]
```diff
@@ -0,0 +1,153 @@
...
+		p.Outputln(string(details))
+		return nil
+
+	default:
+		table := tables.NewTable()
+		table.AddRow("ID", utils.PtrString(backup.Id))
+		table.AddSeparator()
+		table.AddRow("NAME", utils.PtrString(backup.Name))
+		table.AddSeparator()
+		table.AddRow("SIZE", utils.PtrByteSizeDefault((*int64)(backup.Size), ""))
```
Comment (marceljk) [inline internal/cmd/volume/backup/describe/describe.go]: The conversion isn't required because backup.Size is already a *int64 
```suggestion
		table.AddRow("SIZE", utils.PtrByteSizeDefault((backup.Size), ""))
```

File: internal/cmd/volume/backup/list/list.go [hunk 20]
```diff
@@ -0,0 +1,194 @@
...
+			backup := backups[i]
+
+			table.AddRow(
+				utils.PtrString(backup.Id),
+				utils.PtrString(backup.Name),
+				utils.PtrByteSizeDefault((*int64)(backup.Size), ""),
```
Comment (marceljk) [inline internal/cmd/volume/backup/list/list.go]: Same here
```suggestion
				utils.PtrByteSizeDefault((backup.Size), ""),
```

File: internal/cmd/volume/backup/list/list.go [hunk 21]
```diff
@@ -0,0 +1,194 @@
...
+				utils.PtrString(backup.Status),
+				utils.PtrString(backup.SnapshotId),
+				utils.PtrString(backup.VolumeId),
+				utils.PtrString(backup.AvailabilityZone),
+				utils.PtrStringDefault(backup.Labels, ""),
```
Comment (marceljk) [inline internal/cmd/volume/backup/list/list.go]: The labels need to be formatted. see also the output for other commands: https://github.com/stackitcloud/stackit-cli/blob/4f7804790e9f3306c2d3ea770f674239bbaea3b9/internal/cmd/security-group/list/list.go#L145-L151

File: internal/cmd/volume/backup/describe/describe.go [hunk 22]
```diff
@@ -0,0 +1,153 @@
...
+		table.AddSeparator()
+		table.AddRow("STATUS", utils.PtrString(backup.Status))
+		table.AddSeparator()
+		table.AddRow("SNAPSHOT ID", utils.PtrString(backup.SnapshotId))
+		table.AddSeparator()
+		table.AddRow("VOLUME ID", utils.PtrString(backup.VolumeId))
+		table.AddSeparator()
+		table.AddRow("AVAILABILITY ZONE", utils.PtrString(backup.AvailabilityZone))
+		table.AddSeparator()
+		table.AddRow("LABELS", utils.PtrStringDefault(backup.Labels, ""))
```
Comment (marceljk) [inline internal/cmd/volume/backup/describe/describe.go]: The labels need to be formatted. see also the output for other commands: https://github.com/stackitcloud/stackit-cli/blob/4f7804790e9f3306c2d3ea770f674239bbaea3b9/internal/cmd/network-area/describe/describe.go#L207-L212

File: internal/cmd/volume/backup/restore/restore.go [hunk 23]
```diff
@@ -0,0 +1,149 @@
...
+func NewCmd(params *params.CmdParams) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   fmt.Sprintf("restore %s", backupIdArg),
+		Short: "Restores a backup",
+		Long:  "Restores a backup by its ID.",
+		Args:  args.SingleArg(backupIdArg, utils.ValidateUUID),
+		Example: examples.Build(
+			examples.NewExample(
+				`Restore a backup`,
+				"$ stackit volume backup restore xxx-xxx-xxx"),
```
Comment (marceljk) [inline internal/cmd/volume/backup/restore/restore.go]:
```suggestion
				`Restore a backup with ID "xxx"`,
				"$ stackit volume backup restore xxx"),
```

File: internal/cmd/volume/backup/restore/restore.go [hunk 24]
```diff
@@ -0,0 +1,149 @@
...
+			apiClient, err := client.ConfigureClient(params.Printer, params.CliVersion)
+			if err != nil {
+				return err
+			}
+
+			// Get backup details for labels
+			backup, err := apiClient.GetBackup(ctx, model.ProjectId, model.BackupId).Execute()
+			if err != nil {
+				params.Printer.Debug(print.ErrorLevel, "get backup details: %v", err)
+			}
```
Comment (marceljk) [inline internal/cmd/volume/backup/restore/restore.go]: You can move this into a utils function. You can reuse it then also for the update functions, to prompt if the wants to update the backup with the name "yyy"

File: internal/cmd/volume/backup/restore/restore.go (hunk 12 above)
Comment (marceljk) [inline internal/cmd/volume/backup/restore/restore.go]: This line isn't required. You can just use `model.ProjectId` in the `params.Printer.Info()` calls

File: internal/cmd/volume/backup/update/update.go [hunk 25]
```diff
@@ -0,0 +1,170 @@
...
+func NewCmd(params *params.CmdParams) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   fmt.Sprintf("update %s", backupIdArg),
+		Short: "Updates a backup",
+		Long:  "Updates a backup by its ID.",
+		Args:  args.SingleArg(backupIdArg, utils.ValidateUUID),
+		Example: examples.Build(
+			examples.NewExample(
+				`Update a backup name`,
+				"$ stackit volume backup update xxx-xxx-xxx --name new-name"),
```
Comment (marceljk) [inline internal/cmd/volume/backup/update/update.go]:
```suggestion
				`Update the name of a backup with ID "xxx"`,
				"$ stackit volume backup update xxx --name new-name"),
```

File: internal/cmd/volume/backup/update/update.go [hunk 26]
```diff
@@ -0,0 +1,170 @@
...
+			examples.NewExample(
+				`Update backup labels`,
+				"$ stackit volume backup update xxx-xxx-xxx --labels key1=value1,key2=value2"),
```
Comment (marceljk) [inline internal/cmd/volume/backup/update/update.go]:
```suggestion
				`Update the labels of a backup with ID "xxx"`,
				"$ stackit volume backup update xxx --labels key1=value1,key2=value2"),
```

File: internal/cmd/volume/backup/create/create.go (hunk 7 above)
Comment (Benjosh95) [reply internal/cmd/volume/backup/create/create.go]: created a seperate ticket: [https://jira.schwarz/browse/STACKITCLI-203](url)

Comment (rubenhoenle) [reply internal/cmd/volume/backup/create/create.go]: Please use the new util func here after merge of the refactoring PR

File: internal/cmd/volume/backup/create/create_test.go [hunk 27]
```diff
@@ -0,0 +1,305 @@
...
+)
+
+type testCtxKey struct{}
+
+var (
+	testCtx        = context.WithValue(context.Background(), testCtxKey{}, "foo")
+	testClient     = &iaas.APIClient{}
+	testProjectId  = uuid.NewString()
+	testSourceId   = uuid.NewString()
+	testName       = "my-backup"
```
Comment (rubenhoenle) [inline internal/cmd/volume/backup/create/create_test.go:21]: could be const

File: internal/cmd/volume/backup/create/create_test.go [hunk 28]
```diff
@@ -0,0 +1,305 @@
...
+	testLabels     = map[string]string{"key1": "value1"}
+	testSourceType = "volume"
```
Comment (rubenhoenle) [inline internal/cmd/volume/backup/create/create_test.go:22]: same here

File: internal/cmd/volume/backup/create/create_test.go [hunk 29]
```diff
@@ -0,0 +1,305 @@
...
+)
+
+func fixtureFlagValues(mods ...func(flagValues map[string]string)) map[string]string {
+	flagValues := map[string]string{
+		globalflags.ProjectIdFlag: testProjectId,
+		sourceIdFlag:              testSourceId,
+		sourceTypeFlag:            "volume",
```
Comment (rubenhoenle) [inline internal/cmd/volume/backup/create/create_test.go]:
```suggestion
		sourceTypeFlag:            testSourceType,
```
or did I get this wrong? :smile:

File: internal/cmd/volume/backup/create/create_test.go [hunk 30]
```diff
@@ -0,0 +1,305 @@
...
+}
+
+func fixtureInputModel(mods ...func(model *inputModel)) *inputModel {
+	model := &inputModel{
+		GlobalFlagModel: &globalflags.GlobalFlagModel{
+			ProjectId: testProjectId,
+			Verbosity: globalflags.VerbosityDefault,
+		},
+		SourceID:   testSourceId,
+		SourceType: "volume",
```
Comment (rubenhoenle) [inline internal/cmd/volume/backup/create/create_test.go]:
```suggestion
		SourceType: testSourceType,
```

same here

File: internal/cmd/volume/backup/update/update.go [hunk 31]
```diff
@@ -0,0 +1,170 @@
...
+			if err != nil {
+				return err
+			}
+
+			// Configure API client
+			apiClient, err := client.ConfigureClient(params.Printer, params.CliVersion)
+			if err != nil {
+				return err
+			}
+
```
Comment (marceljk) [inline internal/cmd/volume/backup/update/update.go:63]: Here can a prompt be added, to ask the user to confirm this action. Something similar like this:
https://github.com/stackitcloud/stackit-cli/blob/cfa1b8eeace8635863068e5598baf5466c1233bc/internal/cmd/volume/backup/restore/restore.go#L76-L82

File: internal/cmd/volume/backup/create/create.go [hunk 32]
```diff
@@ -0,0 +1,233 @@
...
+			projectLabel, err := projectname.GetProjectName(ctx, params.Printer, params.CliVersion, cmd)
+			if err != nil {
+				params.Printer.Debug(print.ErrorLevel, "get project name: %v", err)
+				projectLabel = model.ProjectId
+			}
+
+			// Get source name for label (use ID if name not available)
+			sourceLabel := model.SourceID
+			if model.SourceType == "volume" {
+				volume, err := apiClient.GetVolume(ctx, model.ProjectId, model.SourceID).Execute()
```
Comment (marceljk) [inline internal/cmd/volume/backup/create/create.go]: Can be replaced with the `GetVolumeName()` function from utils
https://github.com/stackitcloud/stackit-cli/blob/9ba08963822558810f315a6fc29a9edf48d088d8/internal/pkg/services/iaas/utils/utils.go#L61-L67

File: internal/cmd/volume/backup/create/create.go [hunk 33]
```diff
@@ -0,0 +1,233 @@
...
+				if err != nil {
+					params.Printer.Debug(print.ErrorLevel, "get volume name: %v", err)
+				} else if volume != nil && volume.Name != nil {
+					sourceLabel = *volume.Name
+				}
+			} else if model.SourceType == "snapshot" {
+				snapshot, err := apiClient.GetSnapshot(ctx, model.ProjectId, model.SourceID).Execute()
```
Comment (marceljk) [inline internal/cmd/volume/backup/create/create.go]: This can also be replaced with a `GetSnapshotName()` function from utils. Unfortunately we don't have yet the function, so you would need to create it.
Similar to: https://github.com/stackitcloud/stackit-cli/pull/773/files#r2142573897

File: internal/cmd/volume/backup/delete/delete.go [hunk 34]
```diff
@@ -0,0 +1,128 @@
...
+			}
+
+			backup, err := apiClient.GetBackup(ctx, model.ProjectId, model.BackupId).Execute()
+			if err != nil {
+				params.Printer.Debug(print.ErrorLevel, "get backup name: %v", err)
+			}
+			backupLabel := model.BackupId
+			if backup != nil && backup.Name != nil {
+				backupLabel = *backup.Name
+			}
```
Comment (marceljk) [inline internal/cmd/volume/backup/delete/delete.go]: This can be move to a `GetBackupName()` function to iaas utils as well. Because this part of code is used duplicated in the restore command and would be need for https://github.com/stackitcloud/stackit-cli/pull/773/files#r2142564651

File: internal/cmd/volume/backup/delete/delete.go [hunk 35]
```diff
@@ -0,0 +1,125 @@
...
+				s.Start("Deleting backup")
+				_, err = wait.DeleteBackupWaitHandler(ctx, apiClient, model.ProjectId, model.BackupId).WaitWithContext(ctx)
+				if err != nil {
+					return fmt.Errorf("wait for backup deletion: %w", err)
+				}
+				s.Stop()
+			}
+
+			if model.Async {
+				params.Printer.Info("Triggered deletion of backup %q\n", backupLabel)
```
Comment (rubenhoenle) [inline internal/cmd/volume/backup/delete/delete.go]: This prints to stderr (see https://github.com/stackitcloud/stackit-cli/blob/main/internal/pkg/print/print.go#L104-L111) instead of stdout, consider using `Outputf` or `Outputln` (https://github.com/stackitcloud/stackit-cli/blob/main/internal/pkg/print/print.go#L65-L83) instead, they print to stdout. 

This are no error messages, so they don't belong into stderr IMO.

- [x] Check all other outputs (this and your other CLI PRs)

File: go.sum [hunk 36]
```diff
@@ -568,10 +568,10 @@ github.com/stackitcloud/stackit-sdk-go/services/alb v0.5.0 h1:7UKUi7Od7KpuVjV4I/
 github.com/stackitcloud/stackit-sdk-go/services/alb v0.5.0/go.mod h1:RBLBx00zF9MoA/mcLoWwYaACFE0xrWp/EHlzo5S7nhA=
 github.com/stackitcloud/stackit-sdk-go/services/authorization v0.7.0 h1:VpONplkdlEh7Pf22+cNnnHH4bx+S9QI+z55XYRE74JY=
 github.com/stackitcloud/stackit-sdk-go/services/authorization v0.7.0/go.mod h1:dJ19ZwFjp2bfC5ZobXV3vUdSpE3quUw3GuoFSKLpHIo=
-github.com/stackitcloud/stackit-sdk-go/services/dns v0.15.1 h1:+7iIR5r2epGNkAcuERX4kEg1H8hz2yR9+/zLYqZR7Xk=
-github.com/stackitcloud/stackit-sdk-go/services/dns v0.15.1/go.mod h1:xhAdw016dY/hVsLerlExSMocqCc872+S0y1CdV3jAjU=
-github.com/stackitcloud/stackit-sdk-go/services/git v0.5.1 h1:xs8CMY7t8nULQvZr5+XZRs8yWw8YMVw+HfjcuMhieR4=
-github.com/stackitcloud/stackit-sdk-go/services/git v0.5.1/go.mod h1:agI7SONeLR/IZL3TOgn1tDzfS63O2rWKQE8+huRjEzU=
+github.com/stackitcloud/stackit-sdk-go/services/dns v0.15.0 h1:GQAA9gqhKN0ZRc1vRYURHeVjSghh+iF+5DK0HdeuakI=
```
Comment (rubenhoenle) [inline go.sum:571]: Please don't bump service versions unrelated to your changes next time. Dependabot/renovate will handle this for us.

Comment (rubenhoenle) [reply go.sum:571]: Can be left like it is for now.
